injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddSingleElementAnnotation.java

```diff
@@ -0,0 +1,150 @@
+   * passed argument, otherwise, a new annotation will be injected to the node.
+   */
+  private final boolean collapseArguments;
+
+  public AddSingleElementAnnotation(
+      Location location, String annotation, String argument, boolean collapseArguments) {
+    super(location, annotation);
+    Preconditions.checkArgument(
+        argument != null && !argument.equals(""),
+        "argument cannot be null or empty, use AddAnnotation instead.");
+    this.argument = argument;
+    this.collapseArguments = collapseArguments;
+  }
+
+  @Override
+  public void visit(NodeWithAnnotations<?> node) {
+
+    StringLiteralExpr argumentExp = new StringLiteralExpr(argument);
+    // Check all existing annotations with arguments.
+    boolean argumentExists =
+        node.getAnnotations().stream()
+            .anyMatch(
+                annot -> {
+                  if (!(annot.getNameAsString().equals(annotation)
+                      || annot.getNameAsString().equals(annotationSimpleName))) {
+                    return false;
+                  }
+                  if (!(annot instanceof SingleMemberAnnotationExpr)) {
+                    return false;
+                  }
+                  Expression memberExp = ((SingleMemberAnnotationExpr) annot).getMemberValue();
+                  if (memberExp instanceof StringLiteralExpr && memberExp.equals(argumentExp)) {
+                    return true;
+                  }
+                  return memberExp instanceof ArrayInitializerExpr
+                      && ((ArrayInitializerExpr) memberExp).getValues().contains(argumentExp);
+                });
+    if (argumentExists) {
+      return;
+    }
+
+    Optional<AnnotationExpr> annotationWithSameNameExists =
+        node.getAnnotationByName(annotationSimpleName);
+    if (annotationWithSameNameExists.isEmpty()) {
+      // No annotation with this name exists, add it directly.
+      addAnnotationExpressionOnNode(node, argumentExp);
+      return;
+    }
+
+    // Annotation with the same name exists, if collapse is off, add it directly.
+    if (!collapseArguments) {
+      addAnnotationExpressionOnNode(node, argumentExp);
+      return;
+    }
+
+    // Annotation with the same name exists, collapse is on, update it.
+    AnnotationExpr existingAnnotation = annotationWithSameNameExists.get();
+    if (!(existingAnnotation instanceof SingleMemberAnnotationExpr)) {
+      throw new UnsupportedOperationException(
+          "Cannot update existing annotation with type: "
+              + existingAnnotation.getClass()
+              + ". Please file an issue with the expected behaviour at: https://github.com/nimakarimipour/NullAwayAnnotator/issues. Thank you!");
+    }
+
+    SingleMemberAnnotationExpr singleMemberAnnotationExpr =
+        (SingleMemberAnnotationExpr) existingAnnotation;
+    ArrayInitializerExpr updatedMemberValue = new ArrayInitializerExpr();
+    NodeList<Expression> nodeList = new NodeList<>();
+    nodeList.add(argumentExp);
```

Comment:
Nit: I'd move this below the two ifs below, that way we are appending to argument lists, rather than prepending

Reply:
fbc738bbfdfb9570efe8a917b36ba7fcfc24fc91